openedx/openedx-platform
# feat: add SurveyReportUpload and add send report method

### Description:

This PR adds a new model SurveyReportUpload to save the status and details of the send request.

SurveyReport model (https://github.com/openedx/edx-platform/pull/31183).

ISSUE:
https://github.com/openedx/tcril-engineering/issues/349

APPROVED DESIGN:
https://docs.google.com/document/d/1tm1O5BQsAwekphI5IN9pBn8behQDZigXePLGqCl5he8

### DECISIONS:
- We decided add a new model to save all the information about each send request.
- We decided send the survey report data from json to a API endpoint, I hope that tCrill supply this endpoint.
- We decided that each report generated by cli will be send automatically, unless the user set the option `--no-send` in the command.

### How to test
First, You should define the settings: **SURVEY_REPORT_ENDPOINT** with a valid url, you can use some Mock API Service like https://beeceptor.com/.

Then, with a dev environment running,  you should run the command `python manage.py lms generate_report` or `python manage.py lms generate_report --no-send`.
